flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1657,6 +1678,17 @@ IntrinsicLibrary::genChar(mlir::Type type,
   return fir::CharBoxValue{cast, len};
 }
 
+/// LGE, LGT, LLE, LLT
+template <mlir::CmpIPredicate pred>
+fir::ExtendedValue
+IntrinsicLibrary::genCharCompare(mlir::Type type,
+                                 llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 2);
+  return Fortran::lower::genRawCharCompare(
```

Comment:
I think maybe `Raw` is meant to imply an unboxed version here, but `Unboxed` might be a bit clearer.

Reply:
Qualifier `Raw` doesn't really seem necessary.  I'll remove it to make this function an overload of another `GenCharCompare` function.  Doing that triggers some additional names adjustments.